proxy/http2/Http2ConnectionState.cc

```diff
@@ -1601,7 +1595,20 @@ Http2ConnectionState::send_goaway_frame(Http2StreamId id, Http2ErrorCode ec)
   SCOPED_MUTEX_LOCK(lock, this->ua_session->mutex, this_ethread());
   this->ua_session->handleEvent(HTTP2_SESSION_EVENT_XMIT, &frame);
 
-  handleEvent(HTTP2_SESSION_EVENT_FINI, nullptr);
+  this->ua_session->set_half_close_flag(true);
+  DebugHttp2Con(ua_session, "session half-close");
+
+  // As for gracefull shutdown, TS should process outstanding stream as long as possible.
+  // As for signal connection error, TS should close connection immediately.
+  if (ec != Http2ErrorCode::HTTP2_ERROR_NO_ERROR) {
```

Comment:
As far as I understand, the "at least one RTT" is the time to avoid race condition of in-flight stream creation. So TS can process outstanding streams after TS sent second GOAWAY frame.